source/lac/scalapack.cc

```diff
@@ -756,6 +866,412 @@ NumberType ScaLAPACKMatrix<NumberType>::norm(const char type) const
   return res;
 }
 
+
+
+template <typename NumberType>
+void ScaLAPACKMatrix<NumberType>::save(const char *filename) const
+{
+#ifndef DEAL_II_WITH_HDF5
+  (void)filename;
+  AssertThrow(false, ExcMessage ("HDF5 support is disabled."));
+#else
+#  ifdef H5_HAVE_PARALLEL
+  //implementation for configurations equipped with a parallel file system
+  save_parallel(filename);
+
+#  else
+  //implementation for configurations with no parallel file system
+  save_serial(filename);
+
+#  endif
+#endif
+}
+
+
+
+template <typename NumberType>
+void ScaLAPACKMatrix<NumberType>::save_serial(const char *filename) const
+{
+#  ifndef DEAL_II_WITH_HDF5
+  (void)filename;
+  AssertThrow(false, ExcMessage ("HDF5 support is disabled."));
```

Comment:
since it's an internal function which is called from `save()`, i would just do `Assert(false, ExcInternalError())` here.

Reply:
same below for save parallel.